src/main/groovy/net/wasdev/wlp/gradle/plugins/tasks/StartTask.groovy

```diff
@@ -69,14 +67,49 @@ class StartTask extends AbstractServerTask {
                 }
             }
 
-            for (String archiveName : appsToVerify) {
-                String verify = serverTask.waitForStringInLog(START_APP_MESSAGE_REGEXP + archiveName, timeout, serverTask.getLogFile())
-                if (!verify) {
-                    executeServerCommand(project, 'stop', buildLibertyMap(project))
-                    throw new GradleException("The server has been stopped. Unable to verify if the server was started after ${verifyAppStartTimeout} seconds.")
+            def verifyAppStartedThreads = appsToVerify.collect { String archiveName ->
+                Thread.start {
+                    String verify = serverTask.waitForStringInLog(START_APP_MESSAGE_REGEXP + archiveName, timeout, serverTask.getLogFile())
+                    if (!verify) {
+                        executeServerCommand(project, 'stop', buildLibertyMap(project))
+                        throw new GradleException("The server has been stopped. Unable to verify if the server was started after ${verifyAppStartTimeout} seconds.")
+                    }
+                }
+            }
+            verifyAppStartedThreads*.join()
+        }
+    }
+
+    private Set<String> getAppNamesFromServerXml() {
+        Set<String> appNames
+
+        File serverConfigFile = new File(getServerDir(project), 'server.xml')
+        if (serverConfigFile != null && serverConfigFile.exists()) {
+            try {
+                ServerConfigDocument scd = ServerConfigDocument.getInstance(serverConfigFile, server.configDirectory, server.bootstrapPropertiesFile, server.bootstrapProperties, server.serverEnv)
+                if (scd != null) {
+                    appNames = scd.getNames()
+                    appNames += scd.getNamelessLocations().collect { String fileName ->
+                            getNameByFile(fileName)
+                        }
                 }
-                timeout = endTime - System.currentTimeMillis();
+            }
+            catch (Exception e) {
+                logger.warn(e.getLocalizedMessage())
+                logger.debug(e)
+            }
+        }
+        return appNames
+    }
+
+    protected String getNameByFile(String fileName) {
+        String appName = ''
+
```

Comment:
location might have extra stuff at beginning - get down to just the war file name.
for example,
../../../../myapp.war